chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserSubscriptionDispatcherTest.kt

```diff
@@ -0,0 +1,157 @@
+
+    val simpleJoinedRoomApiType = JoinedRoomApiType(
+        id = "id1",
+        name = "room1",
+        createdById = "person1",
+        pushNotificationTitleOverride = "notification override",
+        private = false,
+        customData = mapOf("highlight" to "blue"),
+        lastMessageAt = "2020-02-27T17:12:10Z",
+        updatedAt = "2020-02-27T17:12:20Z",
+        createdAt = "2020-02-27T17:12:30Z",
+        deletedAt = null
+    )
+
+    val simpleUser = UserApiType(
+        id = "user1",
+        createdAt = "",
+        updatedAt = "",
+        name = "name",
+        avatarURL = null,
+        customData = null,
+        online = false)
+
+    describe("given no existing state") {
+        val dispatcher = mockk<Dispatcher>(relaxed = true)
+        val differ = mockk<JoinedRoomsStateDiffer> {
+            every { stateExists() } returns false
+        }
+        val dateApiTypeMapper = DateApiTypeMapper()
+        val joinedRoomApiTypeMapper = JoinedRoomApiTypeMapper(dateApiTypeMapper)
+        val userSubscriptionDispatcher = UserSubscriptionDispatcher(
+            joinedRoomApiTypeMapper = joinedRoomApiTypeMapper,
+            joinedRoomsStateDiffer = differ,
+            dispatcher = dispatcher
+        )
+
+        describe("when I receive an InitialState event") {
+            val event = UserSubscriptionEvent.InitialState(
+                currentUser = simpleUser,
+                rooms = listOf(simpleJoinedRoomApiType),
+                readStates = listOf(RoomReadStateApiType("id1", 1, null)),
+                memberships = listOf(RoomMembershipApiType("id1", listOf()))
+            )
+            userSubscriptionDispatcher.onEvent(event)
+
+            it("then JoinedRoomsReceived is dispatched") {
+                verify(exactly = 1) { dispatcher(JoinedRoomsReceived(
+                    rooms = joinedRoomApiTypeMapper.toRoomInternalTypes(event.rooms),
+                    unreadCounts = joinedRoomApiTypeMapper.toUnreadCounts(event.readStates)
+                )) }
+            }
+        }
+
+        describe("when I receive an AddedToRoomEvent") {
+            val event = UserSubscriptionEvent.AddedToRoomEvent(
+                room = simpleJoinedRoomApiType,
+                readState = RoomReadStateApiType("id1", 1, null),
+                membership = RoomMembershipApiType("id1", listOf())
+            )
+            userSubscriptionDispatcher.onEvent(event)
+
+            it("then JoinedRoom is dispatched") {
+                verify(exactly = 1) { dispatcher(JoinedRoom(
+                    room = joinedRoomApiTypeMapper.toRoomInternalType(event.room),
+                    unreadCount = 1
+                )) }
+            }
+        }
+
+        describe("when I receive a RemovedFromRoomEvent") {
+            val event = UserSubscriptionEvent.RemovedFromRoomEvent("room1")
+            userSubscriptionDispatcher.onEvent(event)
+
+            it("then the dispatcher will send a LeftRoom action") {
```

Comment:
```suggestion
            it("then LeftRoom is dispatched") {
```
[consistency](https://github.com/pusher/chatkit-android/pull/193/files#diff-ae4b04dd619cee6ff385e54b0666e2e0R70)

Reply:
Looks like this is the only one that was missing.